cmd/state-svc/internal/resolver/resolver.go

```diff
@@ -116,64 +116,55 @@ func (r *Resolver) Version(ctx context.Context) (*graph.Version, error) {
 	}, nil
 }
 
-func (r *Resolver) AvailableUpdate(ctx context.Context, channel, version string) (*graph.AvailableUpdate, error) {
+func (r *Resolver) AvailableUpdate(ctx context.Context, desiredChannel, desiredVersion string) (*graph.AvailableUpdate, error) {
 	defer func() { handlePanics(recover(), debug.Stack()) }()
 
+	if desiredChannel == "" {
+		desiredChannel = constants.BranchName
+	}
+
 	r.an.EventWithLabel(anaConsts.CatStateSvc, "endpoint", "AvailableUpdate")
-	logging.Debug("AvailableUpdate resolver")
+	logging.Debug("AvailableUpdate resolver: %s/%s", desiredChannel, desiredVersion)
 	defer logging.Debug("AvailableUpdate done")
 
 	var (
-		update *updater.Update
-		ok     bool
-		err    error
+		avUpdate *updater.AvailableUpdate
+		ok       bool
+		err      error
 	)
 
 	switch {
-	case channel == constants.BranchName && version == "":
-		update, ok = r.updatePoller.ValueFromCache().(*updater.Update)
-		if !ok || update == nil {
+	case desiredChannel == constants.BranchName && desiredVersion == "":
```

Comment:
Nit: This switch case seems pointless considering there's only really one conditional.

Reply:
I tend to prefer switch/case/default over if/else since, imo, it's more readable.